iocore/hostdb/HostDB.cc

```diff
@@ -50,78 +54,214 @@ unsigned int hostdb_ip_stale_interval          = HOST_DB_IP_STALE;
 unsigned int hostdb_ip_timeout_interval        = HOST_DB_IP_TIMEOUT;
 unsigned int hostdb_ip_fail_timeout_interval   = HOST_DB_IP_FAIL_TIMEOUT;
 unsigned int hostdb_serve_stale_but_revalidate = 0;
-unsigned int hostdb_hostfile_check_interval    = 86400; // 1 day
+ts_seconds hostdb_hostfile_check_interval{std::chrono::hours(24)};
 // Epoch timestamp of the current hosts file check.
-ink_time_t hostdb_current_interval = 0;
+ts_time hostdb_current_interval{TS_TIME_ZERO};
 // Epoch timestamp of the last time we actually checked for a hosts file update.
-static ink_time_t hostdb_last_interval = 0;
+static ts_time hostdb_last_interval{TS_TIME_ZERO};
 // Epoch timestamp when we updated the hosts file last.
-static ink_time_t hostdb_hostfile_update_timestamp = 0;
-static char hostdb_filename[PATH_NAME_MAX]         = DEFAULT_HOST_DB_FILENAME;
-int hostdb_max_count                               = DEFAULT_HOST_DB_SIZE;
-char hostdb_hostfile_path[PATH_NAME_MAX]           = "";
-int hostdb_sync_frequency                          = 0;
-int hostdb_disable_reverse_lookup                  = 0;
-int hostdb_max_iobuf_index                         = BUFFER_SIZE_INDEX_32K;
-
-// Verify the generic storage is sufficient to cover all alternate members.
-static_assert(sizeof(HostDBApplicationInfo::allotment) == sizeof(HostDBApplicationInfo),
-              "Generic storage for HostDBApplicationInfo is smaller than the union storage.");
+static ts_time hostdb_hostfile_update_timestamp{TS_TIME_ZERO};
+static char hostdb_filename[PATH_NAME_MAX] = DEFAULT_HOST_DB_FILENAME;
+int hostdb_max_count                       = DEFAULT_HOST_DB_SIZE;
```

Comment:
Should we swallow this ugliness to make sure write access to global vars like hostdb_max_count is limited to HostDB.cc?
```
namespace ts
{
template <typename T, class Tag>
struct GlobalAccess;

template <typename T, class Tag, T Iv = T()>
class Global
{
public:
  Global() : _v{Iv} {}

  using Type = T;

  T operator()() const
  {
    return _v;
  }

  friend struct GlobalAccess<T, Tag>;

private:
  T _v;
};

} // end namespace ts

// in HostDB.h
struct HostDBMaxCountTag {};
using HostDBMaxCount = ts::Global<int, HostDBMaxCountTag>;
HostDBMaxCount hostdb_max_count;

// test
int f()
{
  return hostdb_max_count();
}

// in HostDB.cc
namespace ts
{
template <>
struct GlobalAccess<int, HostDBMaxCountTag>
{
  static int &ref()
  {
    return hostdb_max_count._v;
  }
};
}

// test
void g(int i)
{
  ts::GlobalAccess<int, HostDBMaxCountTag>::ref() = i;
}
```

Reply:
Why would that be better than
```
namespace { int hostdb_max_count = DEFAULT_HOST_DB_SIZE; }
```

Reply:
Yes that's better as long as hostdb_max_count doesn't have to be used outside of HostDB.cc.

Reply:
The ideal of the above is to give read-only access outside of HostDB.cc.